src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -16,6 +19,24 @@ public void setup() {
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        assertFalse(productDao.findProducts().isEmpty());
+    }
+
+    @Test
+    public void testFindProductsNegativePrice() {
+        ProductDao actualProductDao = new ArrayListProductDao();
+        Product product = new Product();
+        product.setPrice(new BigDecimal(-1));
+        actualProductDao.save(product);
+        Assert.assertEquals(productDao.findProducts().size(), productDao.findProducts().size());
+    }
+
+    @Test
+    public void testFindProductZeroStock() {
+        ProductDao actualProductDao = new ArrayListProductDao();
+        Product product = new Product();
+        product.setStock(0);
+        actualProductDao.save(product);
+        Assert.assertEquals(productDao.findProducts().size(), productDao.findProducts().size());
```

Comment:
this assert always pass

Reply:
change to  Assert.assertEquals(productDao.findProducts().size(), actualProductDao.findProducts().size());